ibis/expr/types/maps.py

```diff
@@ -433,11 +446,10 @@ def map(
     keys: Iterable[Any] | Mapping[Any, Any] | ArrayColumn,
     values: Iterable[Any] | ArrayColumn | None = None,
 ) -> MapValue:
-    """Create a [map container object](https://docs.python.org/3/glossary.html#term-mapping).
+    """Create a MapValue.
```

Comment:
Specifically not mentioning the Ibis type is intentional here. The specific expression type doesn't particularly matter for most usage.

Can we leave this and the other mentions of specific types out of the top-level docstrings?

Reply:
Similar to above, I was getting annoyed at these docstrings because they were hiding what they were returning. I need to know where to go look in the reference to see what operations I can actually do on the thing that is returned.

Reply:
What was the reason for explicitly not mentioning the Ibis type? Trying to not expose an implementation detail?

Reply:
I think I was concerned about not having any explanatory references to what a `Value` is. If we had that, and we could link to it, then that would be ideal IMO.

Reply:
My first thought is that Value could actually get decently ignored. I think if in the "datashapes" explanation doc we explained the concrete types of Scalar and Column, then we could just throw a sentence into that glossing over Values:

```
A bit of data in ibis is defined by two concepts at right angles to each other:

- A datatype (string, integer, floating, struct, etc)
- A datashape (scalar or columnar)

See this table:

|              | Scalar        | Column        |
|--------------|---------------|---------------|
| StringValue  | StringScalar  | StringColumn  |
| IntegerValue | IntegerScalar | IntegerColumn |
| StructValue  | StructScalar  | StructColumn  |

For example, and IntegerColumn object inherits from both the Column abstract base class, and the IntegerValue abstract base class.

These two characteristics determine what a bit of data can do:
```

What other concepts of Value are important that you would want to cover?

Reply:
See https://github.com/ibis-project/ibis/pull/8557